splink/find_brs_with_comparison_counts_below_threshold.py

```diff
@@ -0,0 +1,225 @@
+        already_visited (Set[frozenset], optional): Set of visited combinations.
+        results (List[Dict[str, str]], optional): List of results. Defaults to [].
+
+    Returns:
+        List[Dict]: List of results.  Each result is a dict with statistics like
+            the number of comparisons, the blocking rule etc.
+    """
+    if current_combination is None:
+        current_combination = []
+    if already_visited is None:
+        already_visited = set()
+    if results is None:
+        results = []
+
+    if len(current_combination) == len(all_columns):
+        return results  # All fields included, meaning we're at a leaf so exit recursion
+
+    br = _generate_blocking_rule(linker, current_combination)
+    comparison_count = (
+        linker._count_num_comparisons_from_blocking_rule_pre_filter_conditions(br)
+    )
+    row = _generate_output_combinations_table_row(
+        current_combination,
+        br,
+        comparison_count,
+        all_columns,
+    )
+
+    already_visited.add(frozenset(current_combination))
+
+    if comparison_count > threshold:
+        # Generate all valid combinations and continue the search
+        combinations = _generate_combinations(
+            all_columns, current_combination, already_visited
+        )
+        for next_combination in combinations:
+            _search_tree_for_blocking_rules_below_threshold_count(
+                linker,
+                all_columns,
+                threshold,
+                next_combination,
+                already_visited,
+                results,
+            )
+    else:
+        results.append(row)
+
+    return results
+
+
+def find_blocking_rules_below_threshold_comparison_count(
+    linker: "Linker", max_comparisons_per_rule, columns=None
+) -> pd.DataFrame:
+    """
+    Finds blocking rules which return a comparison count below a given threshold.
+
+    In addition to returning blocking rules, returns the comparison count and
+    'complexity', which refers to the number of equi-joins used by the rule.
+
+    Also returns one-hot encoding that describes which columns are __fixed__ by the
+    blocking rule
+
+    e.g. equality on first_name and surname is complexity of 2
+
+    Args:
+        linker (Linker): The Linker object
+        max_comparisons_per_rule (int): Max comparisons allowed per blocking rule.
+        columns: Columns to consider. If None, uses all columns used by the
+            ComparisonLevels of the Linker.
+
+    Returns:
+        pd.DataFrame: DataFrame with blocking rules, comparison_count, and complexity.
+    """
+
+    if not columns:
+        columns = linker._column_names_as_input_columns
+
+    columns_as_strings = []
+
+    for c in columns:
+        if isinstance(c, InputColumn):
+            columns_as_strings.append(c.quote().name())
+        else:
+            columns_as_strings.append(c)
+
+    results = _search_tree_for_blocking_rules_below_threshold_count(
```

Comment:
If no results are found nothing is produced:
```py
from splink.duckdb.duckdb_linker import DuckDBLinker
import pandas as pd

from tests.basic_settings import get_settings_dict

df = pd.read_csv("./tests/datasets/fake_1000_from_splink_demos.csv")
# Loop and concatenate
for _ in range(5):
    df = pd.concat([df, df], ignore_index=True)

linker = DuckDBLinker(
    df,
    get_settings_dict(),
)

from splink.find_brs_with_comparison_counts_below_threshold import find_blocking_rules_below_threshold_comparison_count
find_blocking_rules_below_threshold_comparison_count(linker, 50000)
```

Reply:
Have added an error where nothing can be found